### PR TITLE
Use elm-jsonapi in decoders

### PR DESCRIPTION
**Pull request by ThomasWeiser**

### CODE REVIEW

**Comment by ThomasWeiser** [comment]
Builds on #37, which should be reviewed/merged first.

**Comment by oliverbarnes** [comment]
Cool. It might be clearer to add WIP to the PR titles, btw. I'm adding it now

**Comment by ThomasWeiser** [comment]
Finished, please review.

Some notes:
- `elm-jsonapi` works well.
  - Included resources can be related easily.
  - Code style for decoders changes a bits.
    - Long nested chain of `Result.andThen` and lambdas for extracting related resources. Defined an infix operator to easy this code a bit.
  - Have supplemented some utility functions in a new module `JsonApi.Extra`.
- Split out some more project specific utility functions into a new module `Api.Util`.
- Joined the functionality of e.g. `getProposalCmd` and `getProposal` into one function.
- These API functions (like `getProposal`) can be written in a nice pipeline style.

File: src/Api.elm [hunk 1]
```diff
@@ -68,157 +73,106 @@ facebookAuthUrl =
...
         |> Cmd.map wrapMsg
 
 
-createProposalCmd : ProposalInput -> String -> (Msg -> a) -> Cmd a
-createProposalCmd proposalInput accessToken wrapMsg =
-    postProposal proposalInput accessToken
+createProposal : ProposalInput -> String -> (Msg -> a) -> Cmd a
+createProposal proposalInput accessToken wrapMsg =
+    encodeProposalInput proposalInput
+        |> Api.Util.requestPost newProposalEndpoint
```
Comment (oliverbarnes) [inline src/Api.elm:164]: 👌 I like them util functions

File: src/Api.elm [hunk 2]
```diff
@@ -68,157 +73,106 @@ facebookAuthUrl =
...
-        body =
-            "{\"auth_code\": \"" ++ authCode ++ "\"}"
-
-        requestTask =
-            exchangeAuthCodeForToken body
-    in
-        Task.perform AuthFailed GotAccessToken requestTask
-            |> Cmd.map wrapMsg
+authenticate : String -> (Msg -> a) -> Cmd a
+authenticate authCode wrapMsg =
```
Comment (oliverbarnes) [inline src/Api.elm:142]: Isn't there a convention to append `Cmd` to functions with side-effects?

**Review by oliverbarnes — COMMENTED** [review]
Looking very good, just have the small question about naming convention on commands, will merge after clarifying it

File: src/Api.elm (hunk 2 above)
Comment (ThomasWeiser) [reply src/Api.elm:142]: Just took a quick look at the effects libraries at the front page of http://package.elm-lang.org/

No, there is no such convention there.

**Comment by ThomasWeiser** [comment]
Thanks for reviewing. Before merging maybe you want to decide on the naming question in #37.

File: src/Api.elm (hunk 2 above)
Comment (oliverbarnes) [reply src/Api.elm:142]: I see your point, I may have gotten that from tutes like https://www.elm-tutorial.org/en/03-subs-cmds/03-tasks.html, but come to think of it, likewise appending `Task` to all tasks would be silly too :)

**Pull request merged**